instrumentation/scala-2.13.0/src/main/java/scala/concurrent/impl/Transformation_Instrumentation.java

```diff
@@ -0,0 +1,92 @@
+/*
+ *
+ *  * Copyright 2020 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package scala.concurrent.impl;
+
+import com.newrelic.agent.bridge.AgentBridge;
+import com.newrelic.agent.bridge.Transaction;
+import com.newrelic.api.agent.Segment;
+import com.newrelic.api.agent.Token;
+import com.newrelic.api.agent.Trace;
+import com.newrelic.api.agent.weaver.NewField;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+import com.nr.agent.instrumentation.scala.ScalaUtils;
+import scala.Function1;
+import scala.concurrent.ExecutionContext;
+import scala.util.Try;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.nr.agent.instrumentation.scala.ScalaUtils.scalaFuturesAsSegments;
+
+@Weave(originalName = "scala.concurrent.impl.Promise$Transformation")
+public class Transformation_Instrumentation<T>  {
+    private Try<T> _arg = Weaver.callOriginal();
+    @NewField
+    private AgentBridge.TokenAndRefCount tokenAndRefCount;
+
+    /**
+     * Override the submitWithValue so we can replace it with our wrapped version.
+     * It plays similar role to the 2.12 setter for "value" on the old CallbackRunnable
+     */
+    public final Promise.Transformation submitWithValue(final Try<T> resolved) {
```

Comment:
I might be wrong, but shouldn't we have this annotation here?
```scala
@Trace(async = true, excludeFromTransactionTrace = true)
```
This way the segments are reported and it allows to align the `ProducerConsumerWithFutures.testProducerConsumer ` test with how it looks for 2.12.
Currently it has zero segments reported.

Reply:
Hi, the annotation with async=true should only be necessary when calling Token::link. For the 2.12 instrumentation we had to create a second token in the CallbackRunnable constructor to fix an issue Akka-Http. It's not clear to me the Java/scala.concurrent.impl.CallbackRunnable/value_$eq segments are valuable and ideally they wouldn't be included in traces either, but I'd be interested to hear if you find them useful?

Reply:
I don't believe they are useful.  Until I worked on the scala instrumentation I had no idea that method even existed.  I think most scala users would actually find those segments confusing because they are completely unaware that futures work like that under the hood.

Reply:
I don't have a strong opinion here. If for practical purposes it's better to leave these segments out - I'm totally fine with it. Just pointed out the difference in behavior here.